### PR TITLE
add lock recording feature

### PR DESCRIPTION
**Pull request by adbenitez**
allow to lock recording so you don't have to keep your finger pushing
(adapted from Signal code)

deltachat-2021-03-10-183120

### CODE REVIEW

**Comment by Hocuri** [comment]
For some reason, Signal made the touch area for recording a lot narrower:

Current DC:
Screenshot_20210311-095546
This PR:
Screenshot_20210311-095539_1
Signal:
Screenshot_20210311-095529

Personally, I liked the wider touch area, making it easier to hit the button, so I vote for making it wider again.

**Comment by Hocuri** [comment]
Apart from that, lgtm

**Comment by gerryfrancis** [comment]
> For some reason, Signal made the touch area for recording a lot narrower

I think they did this because a wider touch area was necessary before. Now the user is not forced to tap and hold the finger that precisely anymore. ;)

**Comment by Hocuri** [comment]
It was never necessary to hold the finger in that position; only if you move your finger 2cm to the left, the recording is cancelled. BTW WhatsApp and Telegram both have wide touch areas, too (also, they are higher, but anyway)

File: res/layout/microphone_recorder_view.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        android:contentDescription="@string/audio"
+        android:src="?quick_mic_icon" />
+
+    <LinearLayout
+        android:id="@+id/lock_drop_target"
+        android:layout_width="40dp"
+        android:layout_height="64dp"
+        android:layout_gravity="center"
+        android:layout_marginLeft="-9dp"
+        android:layout_marginStart="-9dp"
```
Comment (Hocuri) [inline res/layout/microphone_recorder_view.xml:24]: This corrects a small shift to the right of the lock drop target.

I have no idea why the shift was there in the first place, though.

**Review by Hocuri — APPROVED** [review]
@r10s said that my additions don't need another approval :upside_down_face: So, lets merge it!

**Pull request merged**